lib/puppet/util/windows/adsi.rb

```diff
@@ -94,6 +94,21 @@ def execquery(query)
       wmi_connection.execquery(query)
     end
 
+    # https://docs.microsoft.com/en-us/dotnet/api/microsoft.powershell.commands.domainrole?view=powershellsdk-1.1.0
+    DOMAIN_ROLES = {
```

Comment:
Agree with @gimmyxd here, these should be explicit constants e.g.

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/service.rb#L16-L43

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/error.rb#L58-L65

Actual ref is - https://docs.microsoft.com/en-au/windows/win32/api/dsrole/ne-dsrole-dsrole_machine_role